agent/agent.go

```diff
@@ -427,34 +428,59 @@ func (a *Agent) handlePolicy(p *policystorage.Policy) {
 		// plugins doing this.
 		action.Canonicalize()
 
-		if !action.IsWithinLimits(p.Min, p.Max) {
+		// If the policy is configured with dry-run:true then we set the
+		// action count to nil so its no-nop. This allows us to still
+		// submit the job, but not alter its state.
+		if val, ok := p.Target.Config["dry-run"]; ok && val == "true" {
+			logger.Info("scaling dry-run is enabled, using no-op task group count",
+				"target_config", p.Target.Config)
+			action.Count = nil
+		}
+
+		// A nil count indicates that the current value shouldn't change,
+		// so just notify the policy's target of this event.
+		if action.Count == nil {
+			logger.Info("no change in count")
+
+			if err = (*targetPlugin).Scale(action, p.Target.Config); err != nil {
+				logger.Error("failed to scale target", "error", err)
+			}
+			continue
+		}
+
+		// Check if this action will not violate the min and max limits set by
+		// the policy.
+		withinLimts, err := action.IsWithinLimits(p.Min, p.Max)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to execute action: %s", err))
+			continue
+		}
+
+		if !withinLimts {
 			logger.Info("next count outside limits",
-				"from", currentCount, "to", action.Count, "min", p.Min, "max", p.Max)
+				"from", currentCount, "to", *action.Count, "min", p.Min, "max", p.Max)
 
 			// Make sure new count value is within [min, max] limits
-			action.CapCount(p.Min, p.Max)
+			err := action.CapCount(p.Min, p.Max)
+			if err != nil {
+				logger.Error(fmt.Sprintf("failed to execute action: %s", err))
+				continue
+			}
 
 			logger.Info("updated count to be within limits",
-				"from", currentCount, "to", action.Count, "min", p.Min, "max", p.Max)
+				"from", currentCount, "to", *action.Count, "min", p.Min, "max", p.Max)
 		}
 
-		if action.Count == currentCount {
+		// If count is not nil, but its value doesn't change, assume that
+		// no action should happen, not even an empty scale event.
```

Comment:
so, i think later on once there's support for .Meta, it may be worth logging an event here, so we could at least see the count that would have been set.

Reply:
I think this would be the most common scenario, which is nothing is happening. Adding a call to the scaling API just to register that nothing happened would be too verbose I think.